File: samples/java/persistence/src/main/java/io/sytac/azure/demos/persistence/DBIdentifiedObject.java
package io.sytac.azure.demos.persistence;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * Container for the application object and its identification
 * @param <T>
 */
@AllArgsConstructor
@Getter
public class DBIdentifiedObject<T> {
    String databaseId;
    T value;

    public <V> DBIdentifiedObject<V> mapTo(V mappedValue) {
        return new DBIdentifiedObject<>(this.databaseId, mappedValue);
    }
}

File: samples/nodejs/src/persistence/DBIdentifiedObject.ts
export class DBIdentifiedObject<T> {
    public readonly databaseId: string;
    public readonly value: T


    constructor(databaseId: string, value: T) {
        this.databaseId = databaseId;
        this.value = value;
    }

    public mapTo<V>(other: V): DBIdentifiedObject<V> {
        return new DBIdentifiedObject<V>(this.databaseId, other)
    }
}